content/blog/2023-04-04-the-pci-device-that-wasnt.md

```diff
@@ -0,0 +1,289 @@
+Title: Dumb bugs: the PCI device that wasn't
+Date: 2023-03-24 00:00:00
+Authors: Russell Currey
+Category: Development
+Tags: linux
+
+I was happily minding my own business one fateful afternoon when I receive the following kernel bug report:
+
+```text
+BUG: KASAN: slab-out-of-bounds in vga_arbiter_add_pci_device+0x60/0xe00
+Read of size 4 at addr c000000264c26fdc by task swapper/0/1
+
+Call Trace:
+dump_stack_lvl+0x1bc/0x2b8 (unreliable)
+print_report+0x3f4/0xc60
+kasan_report+0x244/0x698
+__asan_load4+0xe8/0x250
+vga_arbiter_add_pci_device+0x60/0xe00
+pci_notify+0x88/0x444
+notifier_call_chain+0x104/0x320
+blocking_notifier_call_chain+0xa0/0x140
+device_add+0xac8/0x1d30
+device_register+0x58/0x80
+vio_register_device_node+0x9ac/0xce0
+vio_bus_scan_register_devices+0xc4/0x13c
+__machine_initcall_pseries_vio_device_init+0x94/0xf0
+do_one_initcall+0x12c/0xaa8
+kernel_init_freeable+0xa48/0xba8
+kernel_init+0x64/0x400
+ret_from_kernel_thread+0x5c/0x64
+```
+
+OK, so [KASAN](https://www.kernel.org/doc/html/latest/dev-tools/kasan.html) has helpfully found an out-of-bounds access in `vga_arbiter_add_pci_device()`.  What the heck is that?
+
+
+# Why does my VGA require arbitration?
```

Comment:
@BenjaminGrayNp1 will be mad if you don't make this a ## to ensure there's only a single <h1> on the page

Reply:
blame org-mode's markdown export.  i'll change them all to h2s